src/GraphQL.SystemTextJson/ObjectDictionaryConverter.cs

```diff
@@ -0,0 +1,223 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace GraphQL.SystemTextJson
+{
+    /// <summary>
+    /// A custom JsonConverter for converting into a dictionary of objects of their real underlying type.
+    /// </summary>
+    /// <remarks>
+    /// With thanks to @pekkah who shared this from tanka-graphql.
+    /// </remarks>
+    public class ObjectDictionaryConverter : JsonConverter<Dictionary<string, object>>
+    {
+        private readonly JsonSerializerOptions _internalOptions = new JsonSerializerOptions()
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase
+        };
+
+        public override Dictionary<string, object> Read(ref Utf8JsonReader reader, Type typeToConvert,
+            JsonSerializerOptions options)
+        {
+            using var doc = JsonDocument.ParseValue(ref reader);
+            return ReadDictionary(doc.RootElement);
+        }
+
+        public override void Write(Utf8JsonWriter writer, Dictionary<string, object> value,
+            JsonSerializerOptions options)
+        {
+            JsonSerializer.Serialize(writer, value, _internalOptions);
```

Comment:
Why not `WriteDictionary(writer, value, _internalOptions);` ?

I don’t see private write methods being called at all.

Reply:
The write methods aren't even being used, since I only use this to read out the `variables`. Can just drop all these write methods and make them throw. How does that sound?

Reply:
Either fix this to use private methods (wit fix copy-paste parts) or completely remove ones. At your discretion.

Reply:
If you prefer to delete them, then I would like to leave in the comments or in the error message a mention of the reason why this is done.

Reply:
Done. Throw has an error msg.

Reply:
good